### ISSUE TITLE
conflicting react npm dependencies for fresh installation

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#751]
@thenickcox @bmathews 

On trying to set up a fresh environment, I get an error because of `react-addons-shallow-compare@0.14.8 requires a peer of react@^0.14.8 but none was installed.`

I think this comes from react-trix, which has an dependency of the react-addons-shallow-compare. Since we don't specify the version for that, it installs 0.14.8, which conflicts with our react 0.14.6.

Is there a better strategy we can be using here to proactively keep this app out of dependency hell? Maybe we should go back to using shrinkwrap and document the process of updating dependencies with shrinkwrap?

We probably ought to upgrade all the react components soon. Looks like 15.0 is coming out soon, so maybe it makes sense to wait for that? Although we're probably safe with the point releases without needing to make any changes.

**Comment by bmathews** [comment]
Are you getting an error or just a warning? After `rm -rf node_modules && npm cache clean`, everything works okay for me on `npm v3.8.5` and `node v5.8.0`

`npm WARN react-addons-shallow-compare@0.14.8 requires a peer of react@^0.14.8 but none was installed`

Eitherway, we should be shrinkwrapping. I can get that back in and update the docs.

**Comment by ragesoss** [comment]
I get that as an error. I think it depends on the order in which things get installed, or it possibly relates to the fact that the initially vagrant setup I'm starting from errored out during the first npm install with `npm ERR! Maximum call stack size exceeded`.

**Comment by bmathews** [comment]
I'm guessing that it's related to the failed npm install earlier and/or the unpredictability of vagrant's shared folders (if that is still happening). 

I updated to react 0.14.8, which gets rid of that warning. The changes to react were just bug fixes and the app is still working: https://github.com/facebook/react/blob/master/CHANGELOG.md

I also just commited the npm shrinkwrap and added a bit about it to the `docs/frontend.md`

tl;dr: `npm install something --save` updates the shrinkwrap file, otherwise run `npm shrinkwrap` after everything's squared away.
